tools/test.py

```diff
@@ -127,10 +131,25 @@
                                "Currently set search path: %s"
                        % (toolchain, search_path))
 
+         # boolean, true if test configuration file is for module false if mbed-os interface
+        module_conf = False
+
+        # Assign config file. Precedence: test_config>app_config
+        # TODO: merge configs if both given
+        if options.test_config:
+            config, module_conf = get_test_config(options.test_config, mcu)
+            if not config:
+                args_error(parser, "argument --test-config contains invalid path or identifier")
+        elif not options.app_config:
+            config = TestConfig.get_default_config(mcu)
+        else:
+            config = options.app_config
```

Comment:
Instead of selecting either test config or app_config, would it be possible to merge the two, with app_config keys overriding existing test_config keys? This would be useful for things like WiFi credentials (so we don't have to add every bit of information on the command line)